packages/sources/src/components/tree-view/tree-view.demo.ts

```diff
@@ -80,4 +111,55 @@ export namespace TreeViewDemo {
       throw new Error('Updating items collection encountered an error.');
     });
   }
+
+  /**
+   * Updates all items in a tree view item collection.
+   * @param collection {TreeViewItem[]} the collection
+   * @param updater the updater function
+   */
+  function updateTree(
+    collection: TreeViewItem[],
+    updater: (item: TreeViewItem) => TreeViewItem
+  ): TreeViewItem[] {
+
+    return collection.map(root => {
+      const item = updater(root);
+      return {
+        ...item,
+        items: item.items ? updateTree(item.items, updater) : item.items
+      };
+    });
+  }
+
+  /**
+   * Opens all items in a tree view item collection that has a descendant with selected state.
+   * @param collection {TreeViewItem[]} the collection
+   * @param collapse {boolean} if true items with no selected descendants will be closed
+   */
+  function openItems(
+    collection: TreeViewItem[],
+    collapse: boolean
+  ): number {
+    const flatten = (data: any, level: number) => data ? data.flatMap((item: any) => ([{ level, item}, ...flatten(item.items, level + 1)])) : [];
```

Comment:
Kan `data: any` niet een betere typering krijgen? Ik denk dat we al blij zijn met `data: any[]`, maar kan het niet `TreeViewItem[]` worden?

Reply:
Dan moeten we de tsconfig.json updaten naar `"lib": [..., "es2019", ...],` om `flatMap` te herkennen op een array. Voor nu leek me deze methode zo lokaal dat `any` de fijnste oplossing is.

Reply:
OK, doen we andere keer